doc/Makefile

```diff
@@ -10,7 +10,7 @@ BUILDDIR      = build
 # Internal variables.
 PAPEROPT_a4     = -D latex_paper_size=a4
 PAPEROPT_letter = -D latex_paper_size=letter
-ALLSPHINXOPTS   = -d $(BUILDDIR)/doctrees $(PAPEROPT_$(PAPER)) $(SPHINXOPTS) source
+ALLSPHINXOPTS   = -n -d $(BUILDDIR)/doctrees $(PAPEROPT_$(PAPER)) $(SPHINXOPTS) source
```

Comment:
This has uncovered a whole bunch of "broken references" to types in our numpy-style doc strings - if we can't suppress those somehow  (aside from getting intersphinx working I guess?), we may want to remove this flag. It does show a few actual reference problems in the docs though.

You may need to do a clean build to see the warnings I'm referring to.